fe/src/main/java/org/apache/doris/external/EsRestClient.java

```diff
@@ -49,48 +49,52 @@
             .readTimeout(10, TimeUnit.SECONDS)
             .build();
 
-    private String basicAuth;
-
-    private int nextClient = 0;
+    private Request.Builder builder;
     private String[] nodes;
     private String currentNode;
+    private int currentNodeIndex = 0;
 
     public EsRestClient(String[] nodes, String authUser, String authPassword) {
         this.nodes = nodes;
+        this.builder = new Request.Builder();
         if (!Strings.isEmpty(authUser) && !Strings.isEmpty(authPassword)) {
-            basicAuth = Credentials.basic(authUser, authPassword);
+            this.builder.addHeader(HttpHeaders.AUTHORIZATION, Credentials.basic(authUser, authPassword));
         }
-        selectNextNode();
+        this.currentNode = nodes[currentNodeIndex];
     }
 
-    private boolean selectNextNode() {
-        if (nextClient >= nodes.length) {
-            return false;
+    private void selectNextNode() {
+        currentNodeIndex++;
+        // reroute, because the previously failed node may have already been restored
+        if (currentNodeIndex >= nodes.length) {
+            currentNodeIndex = 0;
         }
-        currentNode = nodes[nextClient++];
-        return true;
+        currentNode = nodes[currentNodeIndex];
     }
 
     public Map<String, EsNodeInfo> getHttpNodes() throws Exception {
         Map<String, Map<String, Object>> nodesData = get("_nodes/http", "nodes");
         if (nodesData == null) {
             return Collections.emptyMap();
         }
-        Map<String, EsNodeInfo> nodes = new HashMap<>();
+        Map<String, EsNodeInfo> nodesMap = new HashMap<>();
         for (Map.Entry<String, Map<String, Object>> entry : nodesData.entrySet()) {
             EsNodeInfo node = new EsNodeInfo(entry.getKey(), entry.getValue());
             if (node.hasHttp()) {
-                nodes.put(node.getId(), node);
+                nodesMap.put(node.getId(), node);
             }
         }
-        return nodes;
+        return nodesMap;
     }
 
-    public String getIndexMetaData(String indexName) {
-        String path = "_cluster/state?indices=" + indexName
-                + "&metric=routing_table,nodes,metadata&expand_wildcards=open";
+    public String getIndexMapping(String indexName) {
```

Comment:
it is better to return the `top` needed, such as above `Map<String, EsNodeInfo> getHttpNodes()`

Reply:
Ok, I will modify it